tensorpipe/util/ringbuffer/producer.h

```diff
@@ -10,6 +10,12 @@
 
 #include <tensorpipe/util/ringbuffer/ringbuffer.h>
 
+#include <algorithm>
+#include <array>
+#include <cstring>
+#include <tuple>
+#include <utility>
```

Comment:
Just one minor nit: could you put the stdlib headers in a block before the first-party headers?

The style guide we're following has these blocks:
- [for .cc files only] the corresponding .h file
- C standard headers
- C++ stdlib headers
- third-party headers
- first-party headers

Thanks!